airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcDestinationHandler.java

```diff
@@ -51,18 +51,12 @@ public Optional<TableDefinition> findExistingTable(final StreamId id) throws Exc
 
   @Override
   public boolean isFinalTableEmpty(final StreamId id) throws Exception {
-    final int rowCount = jdbcDatabase.queryInt(
-        """
-        SELECT row_count
-        FROM information_schema.tables
-        WHERE table_catalog = ?
-          AND table_schema = ?
-          AND table_name = ?
-        """,
-        databaseName,
-        id.finalNamespace(),
-        id.finalName());
-    return rowCount == 0;
+    return !jdbcDatabase.queryBoolean(
+        DSL.select(
+            DSL.field(DSL.exists(
+                DSL.selectOne()
```

Comment:
this seems to be equivalent of `SELECT 1 from schema.table` which will always return 1 isn't it ?

Reply:
it's wrapped in `exists()`. So effectively `select exists(select 1 from <table>)`

Reply:
also, from slack convo - I added an explicit `limit(1)` here. Postgres does it automatically, but might as well be explicit